samza-api/src/main/java/org/apache/samza/system/SystemAdmin.java

```diff
@@ -84,12 +96,24 @@ default void validateStream(StreamSpec streamSpec) throws StreamValidationExcept
   }
 
   /**
-   * Clear the stream described by the spec.
+   * Clear the entire stream described by the spec.
    * @param streamSpec  The spec for the physical stream on the system.
    * @return {@code true} if the stream was successfully cleared.
    *         {@code false} if clearing stream failed.
    */
   default boolean clearStream(StreamSpec streamSpec) {
     throw new UnsupportedOperationException();
   }
+
+  /**
+   * Clear each partition up to (and include) the corresponding offset for all system stream partitions specified in the map
+   *
+   * @param offsets a map from system stream partition to offset
+   * @return {@code true} if the stream was successfully cleared.
+   *         {@code false} if clearing stream failed.
+   */
+  default boolean clearStream(Map<SystemStreamPartition, String> offsets) {
+    return false;
```

Comment:
Good point. We currently only use Kafka for intermediate stream, it should be OK to throw Exception here. I will update the code to do this.

Reply:
I also renamed the name of the newly added method from `clearStream` to `deleteRecords` so that it is more explicit what it does and user can differentiate it from the existing clearStream(), which delete the topic rather than records.